packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/PostProcessing/PostProcessor.cs

```diff
@@ -10,23 +10,22 @@
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.Generator.CSharp.Providers;
 
 namespace Microsoft.Generator.CSharp
 {
-    internal class PostProcessor
+    public class PostProcessor
     {
         private readonly string? _modelFactoryFullName;
         private readonly string? _aspExtensionClassName;
         private readonly HashSet<string> _typesToKeep;
         private INamedTypeSymbol? _modelFactorySymbol;
 
-        public PostProcessor(
-            HashSet<string> typesToKeep,
-            string? modelFactoryFullName = null,
-            string? aspExtensionClassName = null)
+        public PostProcessor(string? aspExtensionClassName = null)
         {
-            _typesToKeep = typesToKeep;
-            _modelFactoryFullName = modelFactoryFullName;
+            _typesToKeep = [..CodeModelPlugin.Instance.TypeFactory.UnionTypes, .. CodeModelPlugin.Instance.TypesToKeep];
+            var modelFactory = ModelFactoryProvider.FromInputLibrary();
```

Comment:
I checked the implementation of this method, this just returns a new instance.
I think we should avoid this but use our created provider.
I kind of think model factory deserves an entry in typefactory, maybe we could add one?

Reply:
I agree we should avoid multiple instances of ModelFacotryProvider, but this should not be the main concern of this PR.
cc @JoshLove-msft

Reply:
could we open an issue to track this?
I would like to say that we should fix it now, because even if we track it, we might forget this part we are assigning a new instance which might lead us to weird issues in the future.